includes/subscriptions/class-wc-payments-subscription-service.php

```diff
@@ -659,12 +704,42 @@ private function get_recurring_item_data_for_subscription( WC_Subscription $subs
 				continue;
 			}
 
-			$data[] = [
+			$item_data = [
 				'metadata'  => [ 'wc_item_id' => $item->get_id() ],
 				'price'     => $this->product_service->get_wcpay_price_id( $product ),
 				'quantity'  => $item->get_quantity(),
 				'tax_rates' => $this->get_tax_rates_for_item( $item, $subscription ),
 			];
+
+			$product_price         = (float) $product->get_price();
+			$product_interval      = (int) WC_Subscriptions_Product::get_interval( $product );
+			$product_period        = WC_Subscriptions_Product::get_period( $product );
+			$inclusive_taxes       = $subscription->get_prices_include_tax() ? array_sum( $item->get_taxes()['total'] ) : 0;
+			$item_total            = floatval( $item->get_total() + $inclusive_taxes ) / $item->get_quantity();
+			$subscription_interval = (int) $subscription->get_billing_interval();
+			$subscription_period   = $subscription->get_billing_period();
```

Comment:
Might be worth factoring out this line item price calculation to a separate method.